src/aoe_ctc.jl

```diff
@@ -0,0 +1,147 @@
+    bin_width_qdrift = get_friedman_diaconis_bin_width(qdrift_e[peak - bin_width_window .< aoe .< peak + bin_width_window])
+
+
+    # get σ before correction
+    # fit peak (with the aoe_two_bck fit function which uses two EMGs)
+    h_before = fit(Histogram, aoe_cut, hist_start:bin_width:hist_end) ### the current values: hist_start -20 and hist_end 3
+    ps_before = estimate_single_peak_stats(h_before)
+    result_before, report_before = fit_single_aoe_compton(h_before, ps_before, fit_func=:aoe_two_bck, pseudo_prior=pseudo_prior, uncertainty=true)
+    @debug "Found Best σ before correction: $(round(result_before.σ, digits=2))"
+
+    # create optimization function
+    function f_optimize_ctc(fct, aoe, qdrift_e, bin_width)
+        # calculate drift time corrected aoe
+        aoe_ctc = aoe .+ PolCalFunc(0.0, fct...).(qdrift_e)
+        # fit peak
+        h = fit(Histogram, aoe_ctc, minimum(aoe_ctc):bin_width:maximum(aoe_ctc))
+        ps = estimate_single_peak_stats(h)
+        result_peak, report_peak = fit_single_aoe_compton(h, ps, fit_func=:aoe_two_bck, pseudo_prior=pseudo_prior, uncertainty=false) ### in ctc.jl uncertainty false is used (but I don't know why)
+        # get σ and peak height
+        μ = mvalue(result_peak.μ)
+        σ = mvalue(result_peak.σ)
+        p_height = maximum(report_peak.f_fit.(μ-0.2*σ:0.001:μ+0.2*σ)) # hardcoded 0.001 for now
+        # use ratio of σ and peak height as optimization functional
+        return log(σ/p_height)
+    end
+
+    # create function to minimize
+    f_minimize = let f_optimize=f_optimize_ctc, aoe=aoe_cut, qdrift_e=qdrift_e_cut, bin_width=bin_width
+        fct -> f_optimize(fct, aoe, qdrift_e, bin_width)
+    end
+
+    # minimize function
+    qdrift_median = median(qdrift_e_cut)
+    # upper bound
+    fct_lb = [(1e-4 / qdrift_median)^(i) for i in 1:pol_order]
+    @debug "Lower bound: $fct_lb"
+    # lower bound
+    fct_ub = [(50.0 / qdrift_median)^(i) for i in 1:pol_order]
+    @debug "Upper bound: $fct_ub"
+    # start value
+    fct_start = [(1 / qdrift_median)^(i) for i in 1:pol_order]
+    @debug "Start value: $fct_start"
+
+    # optimization
+    optf = OptimizationFunction((u, p) -> f_minimize(u), AutoForwardDiff())
+    optpro = OptimizationProblem(optf, fct_start, [], lb=fct_lb, ub=fct_ub)
+    res = solve(optpro, NLopt.LN_BOBYQA(), maxiters = 3000, maxtime=optim_time_limit)
+    converged = (res.retcode == ReturnCode.Success)
+
+    # get optimal correction factor
+    fct = res.u
+    @debug "Found Best FCT: $(fct .* 1e6)E-6"
+
+    if !converged @warn "CTC did not converge" end
+
+    # calculate drift time corrected aoe
+    _aoe_ctc = aoe_all .+ PolCalFunc(0.0, fct...).(qdrift_e_all)
+    
+    # normalize once again to μ = 0 and σ = 1
+    h_norm = fit(Histogram, _aoe_ctc, -50:bin_width:30) ### hard-coded values: should include some tolerance to higher values
```

Comment:
Here, the histogram edges are hard-coded to `-50` and `30`. I remember agreeing on something like `-20` to `3`, but the charge trapping correction introduces a bias towards higher values (that is afterwards normalized), so `3` might be a bit too little. Maybe let this depend on peakstats?